src/cpp/server/server_builder.cc

```diff
@@ -396,6 +410,35 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
     cq->RegisterServer(server.get());
   }
 
+  for (auto& unstarted_listener : unstarted_passive_listeners_) {
+    has_frequently_polled_cqs = true;
+    auto passive_listener = unstarted_listener.passive_listener.lock();
+    auto* core_server = grpc_core::Server::FromC(server->c_server());
+    if (passive_listener != nullptr) {
+      auto* creds = unstarted_listener.credentials->c_creds();
+      if (creds == nullptr) {
```

Comment:
Why do we need this special case?  Can't we just have the C++ `InsecureCredentials` type return the C-core insecure creds object, just like the secure credential types do?

Reply:
The answer is: ownership and lifetimes, but we could change that. InsecureCredentials don't own a ref to the core type today. It seemed better to handle this edge case here, rather than change the cost of using the type for all cases.

Reply:
Alright, I've given InsecureServerCredentials some grpc_server_credentials to manage.

Reply:
I also got rid of the default implementation of c_creds(), so all creds types will have to return something here, no more special cases.